api/server/router/network/network_routes.go

```diff
@@ -70,14 +72,62 @@ func (n *networkRouter) getNetwork(ctx context.Context, w http.ResponseWriter, r
 		return err
 	}
 
-	nw, err := n.backend.FindNetwork(vars["id"])
-	if err != nil {
-		if nr, err := n.cluster.GetNetwork(vars["id"]); err == nil {
-			return httputils.WriteJSON(w, http.StatusOK, nr)
+	// In case multiple networks have duplicate names, return error.
+	// TODO (yongtang): should we wrap with version here for backward compatibility?
+
+	// Find based on full ID
+	nr, _ := n.cluster.GetNetworks()
+	for _, network := range nr {
+		if network.ID == vars["id"] {
```

Comment:
Thanks @aaronlehmann. I take a look at the route of the network endpoint:
https://github.com/docker/docker/blob/48dd90d3985889ca008faa3b041bf31d2ada95c5/api/server/router/network/network.go#L34-L35
It looks like `id` is guided by:
```
		router.NewGetRoute("/networks/", r.getNetworksList),
		router.NewGetRoute("/networks/{id:.+}", r.getNetwork),
```
In case `id` is empty, it will be rerouted to `getNetworksList` instead of `getNetwork`. So I think `vars["id"]` is fine here.